### ISSUE TITLE
`String>>#asInteger` has not defined behavior

### ISSUE THREAD

**Issue by jordanmontt** [guillep/pharo-core#17277]
The comment of the method says the following: 

```st
String >> #asInteger
    "Return the integer present in the receiver, or nil. In case of float, returns the integer part."
    ....
```

But is not clear. Does returns the integer represented by String, what happens if the String is not an integer.

As an example, let's look at this: 

```st
'Weird222String' asInteger. "222"
'Weird222String111' asInteger. "222"
'333Weird222String111' asInteger. "333"
```
If we see into the implementation, the method returns the first characters that can be parsed as an integer and it stops as soon as there is something else.

I propose to use the NumberParser to parse the String and raise an exception if the complete String does not correspond to an integer.

**Comment by astares** [comment]
Funny enough this seems to be the case since Squeak times. I tried in SqueakJS (squeak.js.org).

![image](https://github.com/user-attachments/assets/20b8adec-84c5-4926-beff-3847d5b4d4fa)

not a problem in Cuis or Squeak Mini image as there is no implementation of #toInteger for String

![image](https://github.com/user-attachments/assets/b1a3da4e-75f8-4f4e-9073-cdd0b6715d5a)

**Comment by ScyAge** [comment]
NumberParser when parsing a Float returns a SmallFloat or the definition of asInteger wants to keep only the Integer part.

- we may can round the float to keep the Integer part but I don't think it is a good solution

**Comment by Ducasse** [comment]
I think that NumberParser parseInteger: '1.3' should raise an error so that the user is aware of it. 
Similarly NumberParser parseFloat: '1' should return a float.

**Comment by Ducasse** [comment]
Such conversion methods such asInteger are not good. I would remove them from String

**Comment by jordanmontt** [comment]
Yes, that is also a solution. Remove that from String, that has a lot of methods anyway. And the users should use the NumberParser explicity
